lib/foreman/util.rb

```diff
@@ -22,9 +22,10 @@ def secure_token
       SecureRandom.base64(96).tr('+/=', '-_*')
     end
 
-    # recommended to make encryption_key at least 32 bytes. Ex. SecureRandom.hex(20)
+    # recommended to make encryption_key 32 bytes, matching the key length preferred by
+    # AS::MessageEncryptor's default algorithm
     def secure_encryption_key
-      SecureRandom.hex(20)
+      SecureRandom.hex(ActiveSupport::MessageEncryptor.key_len / 2)
```

Comment:
Please file a new PR or issue for it, simply changing this to Base64 isn't any more correct than hexadecimal (neither provides a full 8-bit 32-byte range) and changing the format of the encryption key is outside of the scope of this ticket.